src/platform/utilities/toggles/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import environments from 'platform/utilities/environment';
+
+function makeEnvironmentToggleValues(env = environments) {
+  return {
+    develop: env.isDev(),
+    localhost: env.isLocalhost(),
+    notProduction: !env.isProduction(),
+    staging: env.isStaging(),
+    production: env.isProduction(),
+  };
+}
+
+const ToggleContext = React.createContext();
+
+function ToggleProvider(props) {
+  const {
+    children,
+    initialToggleValues,
+    subscribeToUpdateToggles = () => {},
+    unsubscribeToUpdateToggles = () => {},
+  } = props;
+
+  const [toggleValues, setToggleValues] = useState(initialToggleValues);
+
+  function handleToggleValuesUpdate(newToggleValues) {
+    setToggleValues(newToggleValues);
+  }
+
+  useEffect(
```

Comment:
This effect isn't used yet. The idea here is that the toggle client provides a callback that enables the `ToggleContext` to subscribe to feature toggle updates. The toggle client (not implemented yet) will be responsible for polling feature flag updates and pushing the new toggle values into the `ToggleContext` via `handleToggleValuesUpdate`. This will trigger a re-render in the `ToggleContext.Consumer`s.